provisioning/Samples/device/SymmetricKeyWithTrustBundleSample/readme.md

```diff
@@ -0,0 +1,132 @@
+
+## Public API
+
+```diff
+{
+     namespace Microsoft.Azure.Devices.Provisioning.Client {
+         public class ProvisioningDeviceClient {
+             public Task<DeviceRegistrationResult> RegisterAsync(ProvisioningRegistrationAdditionalData data, CancellationToken cancellationToken = default(CancellationToken));
+             public Task<DeviceRegistrationResult> RegisterAsync(ProvisioningRegistrationAdditionalData data, TimeSpan timeout);
+         }
+         public class DeviceRegistrationResult {
++           public string TrustBundle { get; set; }
+         }
++		 public class TrustBundle {
++		 	public TrustBundle();
++		 	public List<X509CertificateWithMetadata> Certificates { get; internal set; }
++		 	public DateTime CreatedDateTime { get; internal set; }
++		 	public string Etag { get; internal set; }
++		 	public string Id { get; internal set; }
++		 	public DateTime LastModifiedDateTime { get; internal set; }
++		 }
+	 
++		 public class X509CertificateMetadata {
++		 	public X509CertificateMetadata();
++		 	public string IssuerName { get; internal set; }
++		 	public DateTime NotAfterUtc { get; internal set; }
++		 	public DateTime NotBeforeUtc { get; internal set; }
++		 	public int SerialNumber { get; internal set; }
++		 	public string Sha1Thumbprint { get; internal set; }
++		 	public string Sha256Thumbprint { get; internal set; }
++		 	public string SubjectName { get; internal set; }
++		 }
+	 
++		 public class X509CertificateWithMetadata {
++		 	public X509CertificateWithMetadata();
++		 	public string Certificate { get; internal set; }
++		 	public X509CertificateMetadata Metadata { get; internal set; }
++		 }
+     }
+}
+```
+
+## Sample
+
+The `SymmetricKeyWithTrustBundleSample` demonstrates how to use the public API to make a registration request to DPS and receive the trust bundle certificates in return.
+
+1. Prerequisites
+
+    1. Using curl, submit the following command to the DPS Service API to create the Trust Bundle in DPS. This command uploads a test private root certificate to DPS as a trust bundle.
```

Comment:
Is this step doable using the provisioning service client? Asking users to use curl to create a trust bundle feels a bit odd

Reply:
This isn't available on the provisioning client. From the look of it, I think this might be a control plane operation since it has to do with updating the provisioning service instance itself (similar to how we upload and verify the root certificates for group enrollments on the provisioning service).
Portal and CLI support are expected to be added later on. For now, these curl commands are all that are available.